Pull request to merge from dev-gmx to dev-frontend

Carl26 opened the pull request.

Carl26 committed:
Refactored Signin page

Changed some final variables into field declaration and system.out.print
to log.i

Carl26 committed:
Added reset password

Added the function of reset password

Carl26 committed:
Updated setting page

Updated setting page for easier navigation about what to do

Carl26 committed:
Merge branch 'development-frontend' into dev-gmx

Carl26 commented:
This update only contains the new forgot password feature and some minor improvements

Carl26 committed:
Added trakt signin resources

added icons and button for trakt signin

Carl26 committed:
Fixed invalid color error

Carl26 committed:
Refactored registration page into registration activity

this new activity state enables back from registration to signin by
pressing back button
solved the problem of incorrect implementation of using setContentView
to change main view

Carl26 committed:
Removed unnecessary comment-outs and imports in signin and registration page

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/ResetPasswordActivity.java:
```diff
@@ -0,0 +1,329 @@
+}
+
+class ResetPwdWorkingThread extends AsyncTask<String, Void, Void> {
+    String vCodeUrl = "http://128.199.231.190:8080/api/users/resetPassword";
```
Actually we should make it consistent to use Singleton instance of httpbuilder, easier to maintain.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/res/layout/activity_reset_password.xml:
```diff
@@ -0,0 +1,111 @@
+        android:layout_width="200dp"
+        android:id="@+id/rsETEmail"
+        android:cursorVisible="true"
+        android:textSize="13dp"
```
As mentioned in my pr, textSize should be in sp.

Carl26 committed:
Implemented 3rd party sign up

ThirdPartySignupAcitivity detects whether the user exists
Updated Signin page and registration to accomendate this new feature

Carl26 committed:
Minor formatting fix

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/GlobalReceiver.java:
```diff
@@ -60,6 +60,11 @@ public void onReceive(Context context, Intent intent) {
+                ((Activity) context).finish();
+                break;
+
             case "UserUpdate":
```
to be consistent, this should be 'userUpdate'. Or change others to start with a capital letter.

Sheng-Xuan commented on frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationActivity.java:
```diff
@@ -0,0 +1,125 @@
+        unregisterReceiver(globalReceiver);
+    }
+
+    private int emailChecker(String email) {
```
checker can improve:
1.Only one '@' is in the string.
2.pattern of *.* after @.
